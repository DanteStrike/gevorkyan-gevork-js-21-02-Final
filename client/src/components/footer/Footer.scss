@import "src/assets/mixins";

.footer {
  background-color: var(--primary-background);
  border-top: var(--primary-border);

  &__container {
    @include container;

    display: flex;
    align-items: center;
    justify-content: space-between;
    min-height: 50px;
  }

  &__item {
    display: flex;
    width: 200px;
    font-family: var(--secondary-font-family);
    font-size: 18px;
  }

  &__item--switchers {
    align-items: center;
    justify-content: space-between;
  }
}

@media (max-width: $tabletWidth) {
  .footer {
    &__container {
      flex-direction: column;
      align-items: center;
      justify-content: space-between;
    }

    &__item {
      justify-content: center;
      width: 100%;
      padding: 5px;
      font-size: 16px;
      border-top: var(--primary-border);
    }

    &__item--copyright {
      order: 2;
    }

    &__item--switchers {
      justify-content: space-around;
      order: 1;
    }
  }
}
