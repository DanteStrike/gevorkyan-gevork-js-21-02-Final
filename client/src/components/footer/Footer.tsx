import React from 'react';
import './Footer.scss';
import ThemeSwitcher from '../theme-switcher/ThemeSwitcher';
import LangSelector from '../lang-selector/LangSelector';

function Footer() {
  return (
    <footer className="footer">
      <div className="footer__container">
        <p className="footer__item footer__item--copyright">Delta World &copy; 1970-2077</p>
        <div className="footer__item footer__item--switchers">
          <LangSelector />
          <ThemeSwitcher />
        </div>
      </div>
    </footer>
  );
}

export default React.memo(Footer);
