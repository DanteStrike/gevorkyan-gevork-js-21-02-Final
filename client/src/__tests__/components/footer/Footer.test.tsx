import {shallow} from 'enzyme';
import React from 'react';
import Footer from '../../../components/footer/Footer';

describe(`Component Footer should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
