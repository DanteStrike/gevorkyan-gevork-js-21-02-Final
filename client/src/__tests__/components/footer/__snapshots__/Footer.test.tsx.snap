// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Footer should render correctly snapshot default test 1`] = `
<footer
  className="footer"
>
  <div
    className="footer__container"
  >
    <p
      className="footer__item footer__item--copyright"
    >
      Delta World Â© 1970-2077
    </p>
    <div
      className="footer__item footer__item--switchers"
    >
      <LangSelector />
      <ThemeSwitcher />
    </div>
  </div>
</footer>
`;
